### PR TITLE
Simplify ShowEmailCLIPrompts

### PR DESCRIPTION
**Pull request by Jonas-Karl**
- generic ProvideEmailsService#provideEmails
- add Sent to inbox types

~~In this way we do not need a new class for each MailboxType which makes it way easier to expand. This means this PR might throw us back with the required number of classes and lines of code metric. Because of that I'm not sure if we should merge it.~~

I re-added some classes and lines of code. This was needed because too much logic happend in the CLIPrompt. 

~~The sonarcloud issues can be fixed if #112 was merged~~ - done

### CODE REVIEW

**Comment by Jonas-Karl** [comment]
This is finally ready for review :)

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/email/provide/UCCollectionProvideEmails.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import de.dhbw.karlsruhe.students.mailflow.external.infrastructure.email.parsing.FileMailboxRepository;
+
+/**
+ * As we got many useCases to provide emails, we need a collection to reduce the amount of
+ * parameters
+ *
+ * @author Jonas-Karl
+ */
+public record UCCollectionProvideEmails(
```
Comment (seiferla) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/email/provide/UCCollectionProvideEmails.java:11]: do we still use records or do we make the class final ?
I know a record is the best practise for immutable objects, but i would say that a final class with get, hashcode etc methods will be more code

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/cli/usecases/showemails/ShowEmailContentCLIPrompt.java [hunk 2]
```diff
@@ -18,24 +17,18 @@ public class ShowEmailContentCLIPrompt extends AuthorizedCLIPrompt {
...
 
-  private final MailboxType mailboxType;
-
   public ShowEmailContentCLIPrompt(
       BaseCLIPrompt previousPrompt,
-      Email email,
-      ProvideEmailsUseCase provideEmailsUseCase,
-      AuthUseCase authUseCase,
-      MailboxType mailboxType) {
+      Email email, ProvideEmailsUseCase provideEmailsUseCase, AuthUseCase authUseCase) {
```
Comment (seiferla) [inline src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/cli/usecases/showemails/ShowEmailContentCLIPrompt.java:22]: I could imagine that it isnt formatted correctly

**Review by seiferla — APPROVED** [review]

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/cli/usecases/showemails/ShowEmailContentCLIPrompt.java (hunk 2 above)
Comment (Jonas-Karl) [reply src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/cli/usecases/showemails/ShowEmailContentCLIPrompt.java:22]: You are right!

File: src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/email/provide/UCCollectionProvideEmails.java (hunk 1 above)
Comment (Jonas-Karl) [reply src/main/java/de/dhbw/karlsruhe/students/mailflow/core/application/email/provide/UCCollectionProvideEmails.java:11]: Currently we are still using records. I suggest that we change all records at once in the last few days of this project. For now it helps us in the development process by implicitly creating the getter-methods

**Pull request merged**